src/components/AppWrapper.tsx

```diff
@@ -0,0 +1,217 @@
+import React from 'react'
+import {IInfographicsSection, TInfographics, IInfographicsRow} from '../types/IInfographicsSection'
+import Editor from './Editor';
+import Canvas from './Canvas'
+import {IInfographicsDetails} from '../types/IInfographicsDetails'
+import produce from 'immer'
+
+interface IState {
+    infographicsDetails: IInfographicsDetails,
+    lastUsedId: number,
+    selectedSection: IInfographicsSection | null,
+    columnsCount: number,
+}
+
+class AppWrapper extends React.Component<{}, IState> {
+
+    state: IState = {
+        infographicsDetails: {
+            widthPx: 400, // the default width of infographics could be set in "New infographics" form in the future
+            infographics: [] as TInfographics,
+        },
+        lastUsedId: 0,
+        selectedSection: null,
+        columnsCount: 1,
+    }
+
+    render() {
+        const {infographicsDetails, selectedSection} = this.state
+
+        return (
+            <div className="app-wrapper">
+                <Editor
+                    infographicsDetails={infographicsDetails}
+                    selectedSection={selectedSection}
+                    addSection={this.handleAddRow}
+                    divideSection={this.handleDivideSection}
+                    setWidth={this.handleSetWidth}
+                    setHeight={this.handleSetHeight}
+                />
+                <Canvas
+                    infographicsDetails={infographicsDetails}
+                    selectedSection={selectedSection}
+                    handleToggleSelectSection={this.handleToggleSelectSection}
+                    handleDeleteSection={this.handleDeleteSection}
+                    columnsCount={this.state.columnsCount}
+                />
+            </div>
+        )
+    }
+
+    private handleAddRow = () => {
+        const {infographicsDetails, lastUsedId: id} = this.state
+        const {widthPx, infographics} = infographicsDetails
+
+        const newRow: IInfographicsRow = {
+            heightPx: 250,
+            sections: [
+                {
+                    id,
+                    position: {
+                        x: 0,
+                        y: infographics.length,
+                    },
+                    column: {
+                        start: 1,
+                        end: 2,
+                    },
+                    widthPercent: 100,
+                    isActive: false,
+                }
+            ],
+            columns: 1,
+        }
+
+        this.setState({
+            infographicsDetails: {
+                widthPx,
+                infographics: [...infographics, newRow],
+            },
+            lastUsedId: id + 1,
+        })
+    };
+
+    private handleToggleSelectSection = (section: IInfographicsSection) => {
+        const {selectedSection} = this.state
+        // if already selected section is clicked again, unselect it
+        if (section === selectedSection) {
```

Comment:
lze zmensit na

`this.setState({selectedSection: section !== selectedSection ? section : null})`